src/Mod/BIM/ArchWall.py

```diff
@@ -212,10 +212,23 @@ def setProperties(self, obj):
             obj.ArchSketchData = True
         if not "ArchSketchEdges" in lp:
             obj.addProperty("App::PropertyStringList","ArchSketchEdges","Wall",QT_TRANSLATE_NOOP("App::Property","Selected edges (or group of edges) of the base Sketch/ArchSketch, to use in creating the shape of this Arch Wall (instead of using all the Base Sketch/ArchSketch's edges by default).  Input are index numbers of edges or groups.  Disabled and ignored if Base object (ArchSketch) provides selected edges (as Wall Axis) information, with getWallBaseShapeEdgesInfo() method.  [ENHANCEMENT by ArchSketch] GUI 'Edit Wall Segment' Tool is provided in external SketchArch Add-on to let users to (de)select the edges interactively.  'Toponaming-Tolerant' if ArchSketch is used in Base (and SketchArch Add-on is installed).  Warning : Not 'Toponaming-Tolerant' if just Sketch is used."))
-
+        if not hasattr(obj,"ArchSketchPropertySet"):
+            obj.addProperty("App::PropertyEnumeration","ArchSketchPropertySet","Wall",QT_TRANSLATE_NOOP("App::Property","Select User Defined PropertySet to use in creating variant shape, layers of the Arch Wall with same ArchSketch "))
+            obj.ArchSketchPropertySet = ['Default']
+        if not hasattr(self,"ArchSkPropSetPickedUuid"):  # 'obj.Proxy', 'self' not works ?
+            self.ArchSkPropSetPickedUuid = ''
+        if not hasattr(self,"ArchSkPropSetListPrev"):
+            self.ArchSkPropSetListPrev = []
         self.connectEdges = []
         self.Type = "Wall"
 
+    def dumps(self):  # Supercede Arch.Component.dumps()
```

Comment:
Warning, are you sure the parent class does not define those? When implementing such methods, they should always contain something like:
```python
super().dumps()
```

Reply:
Thanks @yorikvanhavre for bringing this up, I noted loads() in ArchComponent do nothing and most of Arch Objects has nothing in loads() so I thought dumps() in Component can be ignored also.  Any mis-understanding?

```
    def loads(self,state):
        return None
```

Reply:
It indeed does not do anything for now, but if in the future someone needs to implement something there, then there will be a problem because the ArchWall will not work properly. So as a rule, if a function is defined in the base class, it's always best to make the re-implemented function call the base function first.

Reply:
For avoidance of doubt, should also contain below, right?
Thanks.

```
    def loads(self,state):
        super().loads()
        ... 
```

Reply:
Yes! Basically you do that for every new class you add, that already exists in the base class

Reply:
OK, added super() to loads() and dumps(), slightly more complicated to keep everything works.

Reply:
Okay, thanks!!